logger/unilog.go

```diff
@@ -149,6 +153,83 @@ const (
 // Stats is Unilog's statsd client.
 var Stats *statsd.Client
 
+// tagPair is a simple pair of a tag t and the full metric name n
+type tagPair struct {
+	t string
+	n string
+}
+
+// independentTags stores a list of tags to individually emit metrics on.
+// Each metric name will be formatted exactly once and cached in metricsTable.
+type independentTags struct {
+	// The tags to build metric names from
+	// Format is foo:bar where foo is the tag name
+	Tags []string
+	// Lookup table for metricName -> slice of metricName.tagName
+	metricsTable map[string][]tagPair
+}
+
+func newIndependentTags(tags []string) *independentTags {
+	return &independentTags{Tags: tags, metricsTable: make(map[string][]tagPair)}
+}
+
+func setupIndependentTags() *independentTags {
+	return newIndependentTags(strings.Split(independenttags, ","))
+}
+
+func (it *independentTags) GetTags(key string) []tagPair {
+	if it == nil {
+		return []tagPair{}
+	}
+	tags, ok := it.metricsTable[key]
+	if ok {
+		return tags
+	}
+	for _, tag := range it.Tags {
```

Comment:
`tags` doesn't get initialized (its return value is not guaranteed in the error case), so the append() below is possibly not correct. We should set it to empty before the loop.

Reply:
👍